frontend/src/bundles/overview/components/chart-goal-progress/helpers/generate-monthly-datasets.helper.ts

```diff
@@ -0,0 +1,52 @@
+import { type WorkoutResponseDto } from '~/bundles/workouts/types/types.js';
+
+import { type ChartGoalDataset } from '../types/chart-goal-dataset.type.js';
+
+const generateMonthlyDatasets = (
```

Comment:
can we simplify all of these with date-fns?

Reply:
I thought about it, but for the moment cant come up with better approach

Reply:
we have date-fns for it, we dont need new package